File: 0000_0100/9/9.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        long temp = x;
        long rev = 0;
        while (x > 0) {
            rev = rev * 10 + x % 10;
            x /= 10;
        }
        return temp == rev;
    }
};

File: 0000_0100/09/9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        temp = x
        rev = 0
        while (x > 0):
            digit = x % 10
            rev = rev * 10 + digit
            x = x // 10
        return temp == rev


sol = Solution()
print(sol.isPalindrome(1221))
